Route non auth'd users to login with redirect param

jondewijones opened the pull request:
### What

Route non auth'd users to login with redirect param

### How to review

Access app without fake session cookie set and test redirect and param work/look correct

### Who can review

Anyone

franmoore05 commented:
Are we at the point where we could start adding some unit/component tests to this yet?

jondewijones commented:
@franmoore05 i've added Jest so we can run unit tests and added tests for what exist currently in auth.js 

as discussed on our call we'll leave component tests to the "integrate auth" ticket. this goes for the auth middleware that will also get tested by our component tests - that might be the best way to do it, see: https://github.com/vercel/next.js/discussions/38809#discussioncomment-5447811
